app/grandchallenge/core/static/js/formset.mjs

```diff
@@ -1,10 +1,17 @@
-const formsetPrefix = JSON.parse(document.getElementById("formsetPrefix").textContent)
+$(document).ready(function() {
+    const formsetPrefix = JSON.parse(document.getElementById("formsetPrefix").textContent);
+    const emptyForm = document.getElementById(`empty-form-${formsetPrefix}`).innerHTML;
+    const addButton = document.getElementById(`add-form-row-${formsetPrefix}`);
+    const totalForms = document.getElementById(`id_${formsetPrefix}-TOTAL_FORMS`);
 
-$(`.formset_row-${formsetPrefix}`).formset({
-    addText: 'Add another',
-    deleteText: 'Remove',
-    prefix: formsetPrefix,
-    addCssClass: "btn btn-primary my-2",
-    deleteCssClass: "btn btn-danger delete-row mb-2",
-    hideLastAddForm: true,
+    addButton.addEventListener("click", addFormRow);
+
+    function addFormRow(e) {
+        e.preventDefault();
+
+        const formNum = parseInt(totalForms.value) + 1;
+        const newForm = emptyForm.replace(/__prefix__/g, totalForms.value);
```

Comment:
This JS isn't tested, but we have playwrite tests now so maybe it should be added? Otherwise, just checking that the template is in the response and that `__prefix__` is still used by Django is enough. Honestly though, I felt that Django is stable enough that the API on this bit wouldn't change...

Reply:
I also have not tested this with multiple formsets per page. This is the only place that it is used so I think it's okay. Didn't love working with formsets so unsure if we would repeat this pattern. One place where it would be useful is in the Scoring options of the phase where we have people provide awkward JSON objects.

Reply:
I don't have a very strong opinion. I guess a playwright test would be nice to make sure we get notified when things change, and it would be a pretty simple test as well. But I do agree that it is unlikely to change and so also fine if it doesn't have a test. There are other js-snippets in our code base that would benefit more from E2E tests.

Reply:
In the interests of available time I will leave it. I think this is fine for now.